### PR TITLE
PLAT-7246: fix JDBC StringCodec to return byte array by default for V…

### PR DESCRIPTION
**Pull request by oyeliseiev-ua**
…ARBINARY column type

### CODE REVIEW

File: src/benchmark/README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ java -DTEST_PORT=3307 -Duser.country=US -Duser.language=en -jar target/benchmark
 ### Test plan description
 
 **Drivers used for test execution:**
-* singlestore (1.2.5)
+* singlestore (1.2.6-beta0)
```
Comment (pmishchenko-ua) [inline src/benchmark/README.md:30]: probably this shouldn't be changed

File: src/main/java/com/singlestore/jdbc/client/column/StringColumn.java [hunk 2]
```diff
@@ -18,11 +18,7 @@
 import java.math.BigDecimal;
 import java.math.BigInteger;
 import java.math.RoundingMode;
-import java.sql.Date;
-import java.sql.SQLDataException;
-import java.sql.Time;
-import java.sql.Timestamp;
-import java.sql.Types;
+import java.sql.*;
```
Comment (pmishchenko-ua) [inline src/main/java/com/singlestore/jdbc/client/column/StringColumn.java:21]: revert this change?

**Review by pmishchenko-ua — APPROVED** [review]

File: src/main/java/com/singlestore/jdbc/client/column/StringColumn.java [hunk 3]
```diff
@@ -75,8 +75,8 @@ public StringColumn useAliasAsName() {
   }
 
   @Override
-  public String defaultClassname(Configuration conf) {
-    return String.class.getName();
+  public String defaultClassname(final Configuration conf) {
+    return isBinary() ? "byte[]" : String.class.getName();
```
Comment (AdalbertMemSQL) [inline src/main/java/com/singlestore/jdbc/client/column/StringColumn.java:79]: Hmm... 
Is it possible to use `byte[].class.getName()`

File: src/main/java/com/singlestore/jdbc/client/column/StringColumn.java [hunk 4]
```diff
@@ -113,13 +113,18 @@ public int getPrecision() {
...
+    }
     return buf.readString(length.get());
   }
 
   @Override
   public Object getDefaultBinary(final Configuration conf, ReadableByteBuf buf, MutableInt length)
       throws SQLDataException {
-    return buf.readString(length.get());
+    return getDefaultText(conf, buf, length);
   }
```
Comment (AdalbertMemSQL) [inline src/main/java/com/singlestore/jdbc/client/column/StringColumn.java:128]: Hmm....
It looks weird to me that `getDefaultText` and `getDefaultBinary` return the same thing.
It is the same in all other codecs. Do you know what is the purpose of these two methods and why we don't return binary in `getDefaultBinary`?

File: src/main/java/com/singlestore/jdbc/client/column/StringColumn.java [hunk 5]
```diff
@@ -75,8 +75,8 @@ public StringColumn useAliasAsName() {
...
```
Comment (AdalbertMemSQL) [inline src/main/java/com/singlestore/jdbc/client/column/StringColumn.java:75]: What do you think about having a separate Codec for BINARY and VARBINARY?

File: src/main/java/com/singlestore/jdbc/client/column/StringColumn.java (hunk 4 above)
Comment (oyeliseiev-ua) [reply src/main/java/com/singlestore/jdbc/client/column/StringColumn.java:128]: yes, the method call(getDefaultText/getDefaultBinary) is defined by protocol is used(text/binary) , isBinary flag defines column type, so we need this logic for both protocols

File: src/main/java/com/singlestore/jdbc/client/column/StringColumn.java (hunk 5 above)
Comment (oyeliseiev-ua) [reply src/main/java/com/singlestore/jdbc/client/column/StringColumn.java:75]: Do you mean a separate column for these types? These types are all in the same place because they have the same encoding/decoding functionality the difference is only metadata.

File: src/main/java/com/singlestore/jdbc/client/column/StringColumn.java (hunk 3 above)
Comment (oyeliseiev-ua) [reply src/main/java/com/singlestore/jdbc/client/column/StringColumn.java:79]: No) because primitive array encoded differently
